From f3937cabc8c2a796d1df735e1bd9f4f0adbcdb9d Mon Sep 17 00:00:00 2001
From: Yuichi Sugiyama <yuichis@ricsec.co.jp>
Date: Mon, 12 Oct 2026 12:35:21 +0000
Subject: [PATCH] pta: veraison_attestation: label the TA component ARoT

The only software component in the evidence is the hash of the memory
of the calling TA, but its measurement-type says "PRoT". In the PSA
token profile the PRoT (platform root of trust) is the secure firmware
itself, and services running on top of it are ARoT (application root
of trust) components. A TA is the latter, so label the component
"ARoT". This frees "PRoT" for the TEE core, reported by the next
commit.

Reference values that are matched on the measurement-type (the
Veraison PSA scheme keys them on measurement-value plus type) must be
re-provisioned with the new label.

Upstream-Status: Pending
Signed-off-by: Yuichi Sugiyama <yuichis@ricsec.co.jp>
---
 core/pta/veraison_attestation/veraison_attestation.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/pta/veraison_attestation/veraison_attestation.c b/core/pta/veraison_attestation/veraison_attestation.c
index ea66c25..c9144ac 100644
--- a/core/pta/veraison_attestation/veraison_attestation.c
+++ b/core/pta/veraison_attestation/veraison_attestation.c
@@ -33,7 +33,7 @@
 
 #define EAT_PROFILE "http://arm.com/psa/2.0.0"
 #define LIFECYCLE 12288
-#define MEASURMENT_TYPE "PRoT"
+#define TA_MEASUREMENT_TYPE "ARoT"
 
 /* Signer of the boot images when the platform does not say */
 static const uint8_t test_boot_signer_id[SIGNER_ID_LEN] = {
@@ -165,7 +165,7 @@ static TEE_Result cmd_get_cbor_evidence(uint32_t param_types,
 	const char eat_profile[] = EAT_PROFILE;
 	int psa_client_id = 0;
 	int psa_security_lifecycle = LIFECYCLE;
-	const char measurement_type[] = MEASURMENT_TYPE;
+	const char measurement_type[] = TA_MEASUREMENT_TYPE;
 	uint8_t signer_id[SIGNER_ID_LEN] = { };
 	uint8_t psa_instance_id[INSTANCE_ID_LEN] = { };
 	const struct signing_key *skey = NULL;
-- 
2.25.1

